### PR TITLE
Add support for customizing remaining, numeric health check parameters

### PR DESCRIPTION
**Pull request by timoreimann**
The existing, default parameters slightly vary from the defaults that DO LB provides. In order to avoid introducing behavioral changes, we kept them accordingly.

Details:
- Add and parse new annotations.
- Refactor and extend tests.
- Extend service annotation documentation.
- Add example.
- Update change log.

### CODE REVIEW

File: cloud-controller-manager/do/loadbalancers.go [hunk 1]
```diff
@@ -46,6 +46,29 @@ const (
 	// 'service.beta.kubernetes.io/do-loadbalancer-protocol'.
 	annDOHealthCheckProtocol = "service.beta.kubernetes.io/do-loadbalancer-healthcheck-protocol"
 
+	// annDOHealthCheckIntervalSeconds is the annotation used to specify the
+	// number of seconds between between two consecutive health checks. The
+	// value must be between 3 and 300. Defaults to 3.
+	annDOHealthCheckIntervalSeconds = "service.beta.kubernetes.io/do-loadbalancer-healthcheck-check-interval-seconds"
```
[the remark this answers is not in the document]
Comment (timoreimann) [reply cloud-controller-manager/do/loadbalancers.go]: Good point. Let me create a new PR and rebase this one onto it.

[the remark this answers is not in the document]
Comment (timoreimann) [reply cloud-controller-manager/do/loadbalancers.go]: Hmm, I just discovered that the AWS-specific label keys follow a similar schema: https://kubernetes.io/docs/concepts/cluster-administration/cloud-providers/

I will follow up with the Cloud Provider SIG whether's that's intentional or something that in-tree controllers should only use.

[the remark this answers is not in the document]
Comment (timoreimann) [reply cloud-controller-manager/do/loadbalancers.go]: I got [in touch with SIG cloud provider](https://kubernetes.slack.com/archives/C718BPBQ8/p1552908423053200): Apparently, there is no greater effort to move towards local-only prefixes. Even [other out-of-tree providers are using the global prefix](https://github.com/kubernetes/cloud-provider-alibaba-cloud/blob/5863765918a0dd344484e99faed382abeb2aea0f/cloud-controller-manager/options.go#L33).

It seems like `service.beta.kubernetes.io/<provider>-loadbalancer*` has become a de-facto standard at this point. IMHO, it does not make too much sense to try to migrate away from it without having a dedicated effort driven by the SIG.

[the remark this answers is not in the document]
Comment (jcodybaker) [reply cloud-controller-manager/do/loadbalancers.go]: Ok, that makes sense.  Sorry for the run-around on that.

[the remark this answers is not in the document]
Comment (timoreimann) [reply cloud-controller-manager/do/loadbalancers.go]: No worries, I'm glad you raised that point.

**Pull request merged**